pkg/declared/resources.go

```diff
@@ -68,6 +70,17 @@ func (r *Resources) Update(ctx context.Context, objects []client.Object, commit
 
 	// Record the declared_resources metric, after parsing but before validation.
 	metrics.RecordDeclaredResources(ctx, commit, len(newObjects))
+	if r.previousCommit != commit && r.previousCommit != "" {
+		// Reset the stream value of previous commit to 0 to provide correct input
+		// for the Otel Collector metricstransform processor's MAX aggregation.
+		// This is a temporary fix for issue b/321875474, ensuring the metrics
+		// accurately reflect decreases in declared resources. Eventually we should
+		// migrate to otel-collector-go to use the async gauge so that older streams
```

Comment:
> Eventually we should migrate to otel-collector-go to use the async gauge so that older streams won't get updated when there's no longer new values.

Does this mean that with the current gauge metric it will just keep accumulating metrics every time there's a new commit label value and report all of them forever? 

If so, that seems like a pretty problematic memory leak...

Reply:
I would need to print and check the logs of the timeseries getting written out from otel-agent to see how exactly it's keeping all the streams alive, but the outcome was clear http://b/321875474#comment28.

And if it's a consistent refresh, it does sound like a mem leak or metering fraud, which is not going to be fixed with the sunset of Opencensus. This could be a good reason to migrate to Open Telemetry SDK.

Reply:
Oh, right, we're removing the commit label before sending it to GCM, that's why MAX matters here. So it's not expensive for GCM users, because it's aggregated down to one record, but it's probably expensive for Prometheus users. And it's probably a memory leak in the otel-collector and maybe the otel-agent, which buffers them all.

I'm beginning to think that this workaround maybe isn't worth doing, since it's tangential to the actual problem, and we'll have to rip it out later when we actually fix it.

Reply:
It's less risk of a fix for a patch release as we are getting questions from cx.

Reply:
I've attached the TODO b/ in the comment as a reminder.